Add riaknostic to riak builds

bsparrow435 opened the pull request:
This adds riaknostic to the rebar.config and adds make targets for riaknostic so it is grabbed, built, and correctly copied/linked for rel/devrel/stage builds.

bsparrow435 committed:
Add riaknostic to riak builds

jaredmorrow commented:
I'm looking at it now, but for some reason 'make' is not being smart about not running 'deps' for each dev/dev\* on a devrel so 'make devrel' is taking a long time.  I will work on it and will commit to the branch.

jaredmorrow committed:
Fix makefile targets so 'deps' doesn't get called for each devrel

Due to the structure of the targets, `make` couldn't figure
out that `deps` was already called.  This causes `deps` (the
longest operation) to be called each devrel slowing down
the build considerably.
